e2e_tests/scenarios/tests/common/mod.rs

```diff
@@ -132,28 +136,39 @@ pub async fn insert_identity_with_retries(
 pub async fn mined_inclusion_proof_with_retries(
     client: &Client<HttpConnector>,
     uri: &String,
+    chain: &Chain,
     commitment: &Hash,
     retries_count: usize,
     retries_interval: f32,
-) -> anyhow::Result<InclusionProofResponse> {
+) -> anyhow::Result<()> {
     let mut last_res = Err(anyhow!("No calls at all"));
     for _i in 0..retries_count {
         last_res = inclusion_proof(client, uri, commitment).await;
 
         if let Ok(ref inclusion_proof_json) = last_res {
-            if inclusion_proof_json.0.status == Status::Processed(Mined) {
-                break;
+            if let Some(root) = inclusion_proof_json.root {
+                let res = chain
+                    .identity_manager
+                    .method::<U256, RootInfo>("queryRoot", root.into())
```

Comment:
We should put these function in the abi derive - then we'll be able to just call them like methods, e.g. [like here](https://github.com/worldcoin/world-tree/blob/main/tests/full_flow.rs#L96) (except for calls you wouldn't use the `.send()`)

Reply:
done